sycl/include/sycl/detail/backend_traits_level_zero.hpp

```diff
@@ -160,7 +169,7 @@ struct BackendReturn<backend::ext_oneapi_level_zero,
 };
 
 template <> struct BackendReturn<backend::ext_oneapi_level_zero, queue> {
-  using type = ze_command_queue_handle_t;
+  using type = ze_queue_handle_t;
```

Comment:
How would users know what handle (command-queue or command-list) is returned from "get_native" on SYCL queue?

Reply:
If the SYCL queue used for get_native is non-interop queue then the returned object would be the plugin default for commandlists, standard or immediate.
If the SYCL queue had been created through the interop mechanism then the user would have decided what type of queue was created.
Do you see a need for doing get_native on a SYCL queue about which nothing is known?

Reply:
1) I see piextQueueGetNativeHandle always returns ze_command_queue_handle_t, which is just incorrect
2) I think we must provide user a way to known what did we return in the native handle, otherwise they just cannot use it

Reply:
On #1 above, where do you see "piextQueueGetNativeHandle always returns ze_command_queue_handle_t". The way I've defined it in backend_traits_level_zero.hpp, there is a new base class struct _ze_queue_handle_t {}; and L0 command queue and command list struct types are derived from that, so a pointer to the base class can be cast into either, if you know what it is.

Reply:
Ok, I see what you mean in #1 above. I can fix that by returning a bool along with the native handle.

Bigger question is the ABI breakage: what convention to use to add the new interfaces?

Reply:
> Ok, I see what you mean in #1 above. I can fix that by returning a bool along with the native handle.
> 

I suggest first update the https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_backend_level_zero.md and agree on interfaces before implementing them.